### ISSUE TITLE
P2P test in symmetric mode not working

### ISSUE THREAD

**Issue by rafadsm** [ValveSoftware/GameNetworkingSockets#242]
The first peer trying to connect gets this error after the second peer tries to connect:
```
src\steamnetworkingsockets\clientlib\../steamnetworkingsockets_internal.h(767): Assertion Failed: !IsLocked()
Assertion failed: !"TEST FAILED", file GameNetworkingSockets\tests\test_common.cpp, line 46
```
Second peer still trying to connect while first peer crash

i get this error in current branch and release versions [v1.4.1](https://github.com/ValveSoftware/GameNetworkingSockets/releases/tag/v1.4.1) and [v1.4.0](https://github.com/ValveSoftware/GameNetworkingSockets/releases/tag/v1.4.0), but works fine in [v1.3.0](https://github.com/ValveSoftware/GameNetworkingSockets/releases/tag/v1.3.0)

Steps to reproduce: Just execute `test_p2p.py` and will fail in symmetric mode
Im using WebRTC enabled.

**Comment by zpostfacto** [comment]
Hm, thanks for th8s report.  I thought we had a continuous integration step to test that.  Will investigate when I have a chance.

**Comment by JadeVand** [comment]
@rafadsm may I ask which OS you are testing this on? Because I’m getting similar issue on MacOS. But I don’t have any issues when I test this on Windows..

**Comment by zpostfacto** [comment]
I realize this bug is 2 years old, but I believe that I just fixed it in 1b57ccedc006f696a0677898e784df3a6c58f0cc.
